models/tasks.js

```diff
@@ -161,8 +161,14 @@ const fetchUserTasks = async (username, statuses = []) => {
   }
 };
 
-const fetchUserActiveAndBlockedTasks = async (username) => {
-  return await fetchUserTasks(username, ["active", "pending", "blocked"]);
+/**
+ *
+ * @param username { string } : username which will be used to fetch all self tasks
+ * @returns {Promise<tasks>|Array}
+ */
+const fetchSelfTasks = async (username) => {
+  const allTasks = await fetchUserTasks(username);
+  return allTasks.sort((a, b) => a.endsOn - b.endsOn);
```

Comment:
Any reasons to not use Firestore `orderBy()` function for sorting?

Reply:
Lets go with **orderBy()**

Reply:
@shubham-y @Pavangbhat `orderBy()` is a query provided by firestore, since we are using already written function `fetchUserTasks` here, To use orderBy we need to use it in the fetchuserTasks function which I don't think is a good idea since we are using the function at other places too. So I am continuing with sort method.

Reply:
Can we have a second parameter here and based on it we can use **orderBy**

Reply:
done.